packages/store/src/tables/StoreMetadata.sol

```diff
@@ -0,0 +1,198 @@
+  /** Get tableName */
+  function getTableName(uint256 tableId) internal view returns (string memory tableName) {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 0);
+    return string(_blob);
+  }
+
+  /** Get tableName from the specified store */
+  function getTableName(IStore _store, uint256 tableId) internal view returns (string memory tableName) {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = _store.getField(_tableId, _primaryKeys, 0);
+    return string(_blob);
+  }
+
+  /** Set tableName */
+  function setTableName(uint256 tableId, string memory tableName) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    StoreSwitch.setField(_tableId, _primaryKeys, 0, bytes(tableName));
+  }
+
+  /** Push a slice to tableName */
+  function pushTableName(uint256 tableId, string memory _slice) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 0);
+    bytes memory _newBlob = abi.encodePacked(_blob, bytes(_slice));
+    StoreSwitch.setField(_tableId, _primaryKeys, 0, _newBlob);
+  }
+
+  /** Get abiEncodedFieldNames */
+  function getAbiEncodedFieldNames(uint256 tableId) internal view returns (bytes memory abiEncodedFieldNames) {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 1);
+    return bytes(_blob);
+  }
+
+  /** Get abiEncodedFieldNames from the specified store */
+  function getAbiEncodedFieldNames(IStore _store, uint256 tableId)
+    internal
+    view
+    returns (bytes memory abiEncodedFieldNames)
+  {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = _store.getField(_tableId, _primaryKeys, 1);
+    return bytes(_blob);
+  }
+
+  /** Set abiEncodedFieldNames */
+  function setAbiEncodedFieldNames(uint256 tableId, bytes memory abiEncodedFieldNames) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    StoreSwitch.setField(_tableId, _primaryKeys, 1, bytes(abiEncodedFieldNames));
+  }
+
+  /** Push a slice to abiEncodedFieldNames */
+  function pushAbiEncodedFieldNames(uint256 tableId, bytes memory _slice) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+
+    _primaryKeys[0] = bytes32(uint256(tableId));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 1);
+    bytes memory _newBlob = abi.encodePacked(_blob, bytes(_slice));
+    StoreSwitch.setField(_tableId, _primaryKeys, 1, _newBlob);
+  }
```

Comment:
this seems kinda scary for an abi encoded field

Reply:
it totally does, and it shouldn't be used. `string[]` in general is not in a good place rn

Reply:
looks like this exists for both `bytes` and `string` (not arrays, just dynamic length things)

Reply:
it's not exactly difficult, I think I could make that option very quickly
But it's part of the bigger question of configurability vs simplicity. Currently fields are a simple `Record<string, SchemaType>`. Should we make this a shorthand, and allow a field object with various options? I'm kind of leaning yes, but I think I tend to be a bit biased to the configurability side

Reply:
(I have other options in mind that aren't well thought-out yet that could go there too, like enum wrappers)